### PR TITLE
@W-10770550@ Documentation changes related to 3.0 release

### PR DESCRIPTION
**Pull request by rmohan20**
1. Modified installation instructions for v3.x
2. Removed deprecated flags
3. Addressed minor text changes from #614 
4. Added back @jfeingold35's reverted doc changes

### CODE REVIEW

File: docs/_articles/en/v3.x/getting-started/install.md [hunk 1]
```diff
@@ -3,27 +3,40 @@ title: Install Salesforce Code Analyzer Plugin
 lang: en
 ---
 
-## Install the plug-in
+## Install v3.x
```
Comment (rmohan20) [inline docs/_articles/en/v3.x/getting-started/install.md:6]: Modified installation page to suit v3.x.

File: docs/_articles/en/v3.x/whats-new-v3.md [hunk 2]
```diff
@@ -5,24 +5,18 @@ lang: en
 
 ## What’s new in v3.x?
 
-TODO: (Section on changes to existing engines)
+As a starting point, we have upgraded major versions of the eslint family of engines:
+1. eslint from major version 6 to major version 8
+2. typescript-eslint plugin from major version 2 to major version 5
+3. eslint-lwc plugin from major version 0 to major version 1
 
+Another change is that RetireJS would be included as a default engine along with PMD, Eslint, and Eslint-Typescript.
```
Comment (jfeingold35) [inline docs/_articles/en/v3.x/whats-new-v3.md:13]: "...will be included..."

File: docs/_articles/en/v3.x/whats-new-v3.md [hunk 3]
```diff
@@ -5,24 +5,18 @@ lang: en
...
 
+Some deprecated flags have been removed:
+1. `-v/--violations-cause-error` flag on `scanner:run` has been replaced by `-s/--severity-threshold`. 
 
-<!--todo: make this text pop out -->
-## Installing version v3.x
-By default, latest tag gets installed and as of today, it still points to {{ site.data.versions.scanner }}.
-
-To specifically install the latest {{ site.data.versions-v3.scanner }} version:
+There are other exciting changes in the works which will be released as a part of minor releases. Watch out this space for more information!
```
Comment (jfeingold35) [inline docs/_articles/en/v3.x/whats-new-v3.md:18]: "works, which" or "works that".

File: docs/_articles/en/v3.x/architecture/retire-js-engine.md [hunk 4]
```diff
@@ -5,7 +5,7 @@ lang: en
 ## What is RetireJS?
 [RetireJS](https://retirejs.github.io/retire.js/) is an engine that analyzes a project's third-party JavaScript
 dependencies and identifies those that have known security vulnerabilities. It has a lively and responsive community,
-and its database of vulnerable libraries is updated frequently.
+and its database of vulnerable libraries is updated frequently. Starting v3.x, RetireJS is included in the default set of engines.
```
Comment (jfeingold35) [inline docs/_articles/en/v3.x/architecture/retire-js-engine.md:8]: "Starting with v3.0.0"?

File: docs/_articles/en/v3.x/getting-started/install.md [hunk 5]
```diff
@@ -3,27 +3,40 @@ title: Install Salesforce Code Analyzer Plugin
...
 
 Like all the Salesforce CLI plug-ins, you install the Salesforce Code Analyzer CLI plug-in with one simple step. NOTE: Be sure you've completed the [prerequisites](.v3.x/getting-started/prerequisites/) before you install this plug-in.
 
 
-```bash
-$ sfdx plugins:install @salesforce/sfdx-scanner
-Installing plugin @salesforce/sfdx-scanner...
-installed v{{ site.data.versions-v3.scanner }} 
-```
+By default, latest tag gets installed and as of today, it still points to {{ site.data.versions.scanner }}. You can install the pilot version by pointing to the `latest-pilot` label. Also, note that automatic plugin upgrades don't work here.
```
Comment (jfeingold35) [inline docs/_articles/en/v3.x/getting-started/install.md:11]: Should "latest" be code-formatted, i.e. \`latest\` so it reads as `latest`?

**Review by jfeingold35 — APPROVED** [review]

**Pull request merged**